### PR TITLE
Refine phase separation in module internalization

### PR DESCRIPTION
**Pull request by maximedenes**
For #15409, we need a module internalization phase that does not require term typing.

This PR splits this internalization in two phases.

### CODE REVIEW

**Review by gares — APPROVED** [review]
It could help to let us know why/how the second commit is needed

**Comment by maximedenes** [comment]
> It could help to let us know why/how the second commit is needed

Right, the way I cut it doesn't seem to make a lot of sense. I'll review that and add a third commit which uses the types or remove the second one.

**Comment by maximedenes** [comment]
@gares I removed the second commit which didn't make a lot of sense on its own. The PR now really reflects the phase separation. The point of this separation is that the `intern` part will be executing as part of the syntactic interpretation phase, where `interp` will take place only at typing time.

**Comment by gares** [comment]
vscoq ci job is probably waiting for an overlay to be merged.

**Comment by gares** [comment]
Please rebase, and then I'll merge.

**Comment by maximedenes** [comment]
I just reran, looks ok now. refman failure is spurious, isn'it it?

File: interp/modintern.mli
Comment (ejgallego) [inline interp/modintern.mli:37]: In general I'd be nice if you could document the newly added functions

Comment (maximedenes) [reply interp/modintern.mli:38]: Sure, I added a comment.

**Comment by maximedenes** [comment]
@gares anything still holding this?

**Comment by gares** [comment]
I was waiting for CI, dunno if it got restarted on bedrock2

**Comment by ppedrot** [comment]
@gares I restarted it and it's green.

**Review by ppedrot — APPROVED** [review]
Have a formal kernel maintainer green light.

**Comment by gares** [comment]
@coqbot merge now

**Comment by gares** [comment]
We are all looking forward to episode 2

**Pull request merged**
